src/main/scala/scorex/util/serialization/Serializer.scala

```diff
@@ -0,0 +1,6 @@
+package scorex.util.serialization
+
+trait Serializer[TFamily, T <: TFamily, R <: Reader, W <: Writer] {
+  def serialize(obj: T, w: W): Unit
+  def parse(r: R): TFamily
```

Comment:
Let's make some helper method with `Try[T]` return type like `parseTry` we have discussed